### PR TITLE
[Added] scrollsToBottomOnKeyboardDidBeginEditing property

### LINKED ISSUE #203: Flags to perform automatic scrolling
### Flags to perform automatic scrolling
It would be nice to have two flags for users to control scrolling:
```Swift
var scrollToBottomOnKeyboardBeginsEditing = false
var scrollToBottomOnFirstLoad = true
```

@cwalo I think if we put the `messagesCollectionView.scrollToBottom()` call inside of the `isFirstLayout` flag in `viewDidLayoutSubviews` we don't have to do anything with calculating the `contentOffset`?

### PR DESCRIPTION
**Pull request by SD10**
This closes #203 && #165

### CODE REVIEW

File: Sources/MessagesViewController.swift [hunk 1]
```diff
@@ -240,15 +243,23 @@ extension MessagesViewController: UICollectionViewDataSource {
 
 extension MessagesViewController {
 
+
     fileprivate func addKeyboardObservers() {
         NotificationCenter.default.addObserver(self, selector: #selector(handleKeyboardDidChangeState), name: .UIKeyboardWillChangeFrame, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(handleTextViewDidBeginEditing), name: .UITextViewTextDidBeginEditing, object: messageInputBar.inputTextView)
```
Comment (Sherlouk) [inline Sources/MessagesViewController.swift:258]: Out of interest, why are you choosing to add an observer here over assigning the delegate of the textView?

File: Sources/MessagesViewController.swift [hunk 2]
```diff
@@ -240,15 +243,23 @@ extension MessagesViewController: UICollectionViewDataSource {
...
     }
 
     fileprivate func removeKeyboardObservers() {
         NotificationCenter.default.removeObserver(self, name: .UIKeyboardWillChangeFrame, object: nil)
```
Comment (Sherlouk) [inline Sources/MessagesViewController.swift:262]: Do we need to remove the notification `.UITextViewTextDidBeginEditing` here?

Comment (SD10) [reply Sources/MessagesViewController.swift:262]: Only for some versions of iOS. Good catch 😉

File: Sources/MessagesViewController.swift (hunk 1 above)
Comment (SD10) [reply Sources/MessagesViewController.swift:258]: I thought it would be better to not reserve making the delegate our controller

**Pull request merged**